subprojects/core/src/main/java/org/gradle/configuration/project/LifecycleProjectEvaluator.java

```diff
@@ -76,7 +77,13 @@ private void notifyAfterEvaluate(ProjectEvaluationListener listener, ProjectInte
         } catch (Exception e) {
             if (state.hasFailure()) {
                 // Just log this failure, and pass the existing failure out in the project state
-                LOGGER.error("Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.", e);
+                boolean logStackTraces = project.getGradle().getStartParameter().getShowStacktrace() != ShowStacktrace.INTERNAL_EXCEPTIONS;
+                String infoMessage = "Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.";
+                if (logStackTraces) {
+                    LOGGER.error(infoMessage, e);
+                } else {
+                    LOGGER.error(infoMessage + "\n> {}", e.getMessage() != null ? e.getMessage() : e.getClass().getSimpleName());
```

Comment:
In most cases the secondary failure is *because of* the primary failure. So once you fix the primary one, the other will disappear as well. That's why I think it's good to hide it.